src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/UserServiceTest.kt

```diff
@@ -79,6 +81,63 @@ class UserServiceTest {
   @Nested
   inner class GetExistingUserOrCreate {
 
+    @Test
+    fun `getExistingUserOrCreate with username param calls overloaded function with second parameter`() {
+      val username = "SOMEPERSON"
+
+      val user = UserEntityFactory()
+        .withDefaults()
+        .produce()
+
+      every { mockUserRepository.findByDeliusUsername(username) } returns user
+
+      assertThat(userService.getExistingUserOrCreate(username)).isEqualTo(user)
+      verify(exactly = 1) { userService.getExistingUserOrCreate(username, false) }
+    }
+
+    @Test
+    fun `getExistingUserOrCreate with invalid username does not throw error when dontThrowExceptionOnStaffRecordNotFound set true`() {
```

Comment:
nice! i think the test could be renamed to reflect that it's about staff record not existing, for the code to get to this point the username would be valid. (other test names may need tweaking too)